ui/app/mainui/Popups.qml

```diff
@@ -477,9 +477,49 @@ QtObject {
             id: importCommunitiesPopupComponent
             ImportCommunityPopup {
                 store: root.communitiesStore
+                onJoinCommunity: {
+                    Global.openPopup(requestToJoinCommunityDialog, {
+                        communityId: communityId,
+                        name: communityDetails.name,
+                        imageSrc: communityDetails.image,
+                        introMessage: communityDetails.introMessage
+                    });
+                    close();
+                }
                 onClosed: {
-                    destroy()
+                    destroy();
+                }
+            }
+        },
+
+        Component {
+            id: requestToJoinCommunityDialog
+            RequestToJoinCommunity {
+                id: requestToJoinCommPopup
+                property var dummyAddresses: ["a","b","c","d","e","f","g","h"]
+                loginType: root.rootStore.loginType
+                onSelectAddressesToShare: {
+                    //root.rootStore.requestToJoinCommunityWithAuthenticationWithSharedAddresses(communityId, root.rootStore.userProfileInst.name, dummyAddresses);
+                    //root.communitiesStore.spectateCommunity(communityId);
```

Comment:
I guess this is going to be implemented as part of another ticket?

Reply:
@jrainville this was added and commented by you I guess? I just gave it a try and got 
`Warning: qrc:/app/AppLayouts/stores/RootStore.qml:184: Error: Insufficient arguments (qrc:/app/AppLayouts/stores/RootStore.qml:184, )`
is it implemented?

Reply:
This needs to be fixed for sure; the backend is there to (request to) join a community with shared addresses selection, it's already used in the intro dialog

Reply:
> @jrainville this was added and commented by you I guess? I just gave it a try and got `Warning: qrc:/app/AppLayouts/stores/RootStore.qml:184: Error: Insufficient arguments (qrc:/app/AppLayouts/stores/RootStore.qml:184, )` is it implemented?

Looking at the blame, it says I wrote that, but it's your commit. I think it's because I cherry-picked that it says I did it.

Anyway, it's no big deal, we can hook it up later

Reply:
I don't remember doing this :D I left those signals blank and then I saw code when pulling again. should I add a TODO or something?

Reply:
Yeah I guess that would work